x-pack/plugins/canvas/types/canvas.ts

```diff
@@ -37,12 +37,19 @@ export interface CanvasPage {
   groups: CanvasGroup[];
 }
 
+export interface CanvasVariable {
+  name: string;
+  value: boolean | number | string;
+  type: 'boolean' | 'number' | 'string';
+}
+
 export interface CanvasWorkpad {
   '@created': string;
   '@timestamp': string;
   assets: { [id: string]: CanvasAsset };
   colors: string[];
   css: string;
+  variables: CanvasVariable[];
```

Comment:
Will this change cause older workpads to blow up?  Or do we have a migration step on import?

Reply:
So I tested this and spoke to Corey about it:

When retrieving a workpad with workpad service, we "shim" workpads that are missing the new properties and assign sane defaults (we're doing this for css and variables at the moment) so workpads are automatically "upgraded".

For workpads that have been exported from old releases, if we import them via the workpad loader, we attempt to shim it there if it's missing before we send it to the server to create the workpad saved object. If you are importing via saved objects, it will work "out of the box" since our workpad schema validation isn't applied on saved object importing and then when you view the workpad, once again, the get method shims the missing properties for you

Reply:
Attempting to import a workpad with variables on older versions will just simply drop the variables field from the workpad but it will still succeed